### PR TITLE
Maintenance (Github Actions, CodeSniffer, Phpstan 0.12)

### PR DESCRIPTION
**Pull request by f3l1x**
- Travis -> Github Actions
- Branch-alias to 6.6.x
- Funding
- Introduce makefile

### CODE REVIEW

File: .github/FUNDING.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+github: [f3l1x]
+custom: ["https://contributte.org/partners.html"]
```
Comment (paveljanda) [inline .github/FUNDING.yml:2]: Sorry, I do not aggree with that. Pls have a look at at my comments here: https://github.com/contributte/datagrid-nette-database-data-source/pull/18#issuecomment-742332988
Thanks for understanding

File: composer.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
 	"name": "ublaboo/datagrid",
 	"type": "library",
 	"description": "DataGrid for Nette Framework: filtering, sorting, pagination, tree view, table view, translator, etc",
-	"keywords": ["datagrid", "grid", "nette"],
+	"keywords": ["datagrid", "grid", "nette", "contributte", "tables", "data"],
```
Comment (paveljanda) [inline composer.json:5]: Optional: maybe only "table" (<- tables)?

File: src/Column/ColumnNumber.php [hunk 3]
```diff
@@ -34,12 +34,7 @@ public function getColumnValue(Row $row)
 			return $value;
 		}
 
-		return number_format(
-			(float) $value,
-			(int) $this->numberFormat[0],
-			(string) $this->numberFormat[1],
-			(string) $this->numberFormat[2]
-		);
+		return number_format((float) $value, (int) $this->numberFormat[0], (string) $this->numberFormat[1], (string) $this->numberFormat[2]);
```
Comment (paveljanda) [inline src/Column/ColumnNumber.php:37]: Please revert this line - 135 characters on 1 line seams too long for me. Thanks 👍

File: src/Column/ColumnStatus.php [hunk 4]
```diff
@@ -164,8 +164,6 @@ public function getTemplateVariables(): array
 
 	public function setReplacement(array $replacements): Column
 	{
-		throw new DataGridColumnStatusException(
-			'Cannot set replacement for Column Status. For status texts replacement use ->setOptions($replacements)'
-		);
+		throw new DataGridColumnStatusException('Cannot set replacement for Column Status. For status texts replacement use ->setOptions($replacements)');
```
Comment (paveljanda) [inline src/Column/ColumnStatus.php:167]: Please revert this line - 148 characters on 1 line seams too long for me. Thanks 👍

File: src/ColumnsSummary.php [hunk 5]
```diff
@@ -105,12 +90,7 @@ public function render(string $key): ?string
 			return '';
 		}
 
-		return number_format(
-			$this->summary[$key],
-			$this->format[$key][0],
-			$this->format[$key][1],
-			$this->format[$key][2]
-		);
+		return number_format($this->summary[$key], $this->format[$key][0], $this->format[$key][1], $this->format[$key][2]);
```
Comment (paveljanda) [inline src/ColumnsSummary.php:93]: Please revert this line - Having 4 function arguments on line line may be not well readable for some people. Thanks 👍

File: src/DataSource/DibiFluentDataSource.php [hunk 6]
```diff
@@ -46,7 +46,7 @@ public function __construct(Fluent $dataSource, string $primaryKey)
 
 
 	/********************************************************************************
-	 *                          IDataSource implementation                          *
+	 *                          IDataSource implementation *
```
Comment (paveljanda) [inline src/DataSource/DibiFluentDataSource.php:49]: Please revert, it is not symmetric this way :D

File: src/DataSource/DibiFluentDataSource.php [hunk 7]
```diff
@@ -60,7 +60,9 @@ public function getCount(): int
 	 */
 	public function getData(): array
 	{
-		return $this->data ?: $this->dataSource->fetchAll();
+		return $this->data !== []
+			? $this->data
+			: $this->dataSource->fetchAll();
```
Comment (paveljanda) [inline src/DataSource/DibiFluentDataSource.php:65]: Could you please keep the ternary operator on one line? Is there a problem in a coding standard point of view?

File: src/DataSource/DibiFluentMssqlDataSource.php [hunk 8]
```diff
@@ -30,7 +30,7 @@ public function __construct(Fluent $dataSource, string $primaryKey)
 
 
 	/********************************************************************************
-	 *                          IDataSource implementation                          *
+	 *                          IDataSource implementation *
```
Comment (paveljanda) [inline src/DataSource/DibiFluentMssqlDataSource.php:33]: Same as above, pls revert, thanks

File: src/DataSource/DoctrineCollectionDataSource.php [hunk 9]
```diff
@@ -53,7 +53,7 @@ public function __construct(Collection $collection, string $primaryKey)
 
 
 	/********************************************************************************
-	 *                          IDataSource implementation                          *
+	 *                          IDataSource implementation *
```
Comment (paveljanda) [inline src/DataSource/DoctrineCollectionDataSource.php:56]: Same as above ☝️

File: src/DataSource/DoctrineDataSource.php [hunk 10]
```diff
@@ -72,7 +72,7 @@ public function getQuery(): Query
 
 
 	/********************************************************************************
-	 *                          IDataSource implementation                          *
+	 *                          IDataSource implementation *
```
Comment (paveljanda) [inline src/DataSource/DoctrineDataSource.php:75]: We can change that to some simple doc block....

**Review by paveljanda — CHANGES_REQUESTED** [review]

**Comment by pkristian** [comment]
hello, when this PR (also resolving php 8) will be due? i need it for app update of prod app... reasonable :dollar:  for increased attention possible :)

File: composer.json (hunk 2 above)
Comment (f3l1x) [reply composer.json]: Solved

File: src/Column/ColumnNumber.php (hunk 3 above)
Comment (f3l1x) [reply src/Column/ColumnNumber.php]: Solved

File: src/ColumnsSummary.php (hunk 5 above)
Comment (f3l1x) [reply src/ColumnsSummary.php]: Solved

File: src/DataSource/DibiFluentDataSource.php (hunk 6 above)
Comment (f3l1x) [reply src/DataSource/DibiFluentDataSource.php]: Solved

File: src/DataSource/DibiFluentDataSource.php (hunk 7 above)
Comment (f3l1x) [reply src/DataSource/DibiFluentDataSource.php]: Solved

File: src/DataSource/DibiFluentMssqlDataSource.php (hunk 8 above)
Comment (f3l1x) [reply src/DataSource/DibiFluentMssqlDataSource.php]: Solved

File: src/DataSource/DoctrineCollectionDataSource.php (hunk 9 above)
Comment (f3l1x) [reply src/DataSource/DoctrineCollectionDataSource.php]: Solved

File: src/DataSource/DoctrineDataSource.php (hunk 10 above)
Comment (f3l1x) [reply src/DataSource/DoctrineDataSource.php]: Solved

**Comment by f3l1x** [comment]
Fixes applied. :-)

**Comment by f3l1x** [comment]
> hello, when this PR (also resolving php 8) will be due? i need it for app update of prod app... reasonable 💵 for increased attention possible :)

We would love to see you in your patron's list (https://contributte.org/partners.html). At this moment, most easier way is to subscribe to github sponsors (https://github.com/sponsors/f3l1x).

**Comment by f3l1x** [comment]
Also christmas banner included.

**Before**

before-2 (1)

**After**

after-2 (1)

**Comment by paveljanda** [comment]
Thanks!

**Comment by pkristian** [comment]
> > hello, when this PR (also resolving php 8) will be due? i need it for app update of prod app... reasonable dollar for increased attention possible :)
> 
> We would love to see you in your patron's list (https://contributte.org/partners.html). At this moment, most easier way is to subscribe to github sponsors (https://github.com/sponsors/f3l1x).

made small endosement, sadly PHP8 is not still flawless, see #935
Merry christmas! :)

**Comment by f3l1x** [comment]
It should be fixed https://github.com/contributte/datagrid/releases/tag/v6.6.0

**Pull request merged**